linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml

```diff
@@ -96,14 +310,31 @@
   loop_control:
     loop_var: mac
   register: extract_ip_address_result
+  when: uri_hostname == 'localhost'
+
+- name: "mac_and_ip | wait for dhcp ip address"
+  shell: |
+    until arp -an | grep -q -F {{ extract_mac_address_result.results[mac].stdout }}; do
+      sleep 1
+    done
+    arp -an | grep -F {{ extract_mac_address_result.results[mac].stdout }} | cut -f 2 -d "(" | cut -f 1 -d ")"
+  with_items:
+    - "{{ res_count.stdout }}"
+  loop_control:
+    loop_var: mac
+  register: extract_ip_address_result_remote
+  remote_user: "{{ res_def['remote_user'] | default('root') }}"
+  delegate_to: "{{ uri_hostname }}"
```

Comment:
Why is delegate_to necessary here when the uri_hostname is localhost?

Reply:
Noted 
delegate_to is required when uri_hostname is not equal to localhost because , 
the ip addresses are local to machine libvirt is running . We need to get inside the machine to extract machine ip from arp command .